shim-array.js

```diff
@@ -53,6 +53,17 @@ Array.unzip = function (table) {
 };
 
 function define(key, value) {
+
+    if (Array.prototype.hasOwnProperty(key)) {
+        if (
+            typeof console !== 'undefined' &&
+                typeof console.warn === 'function'
+        ) {
+            console.warn('Overrided Array.prototype.' + key, "Saved into _" + key);
```

Comment:
It might be worthwhile to make a `warn` function that isolates the concern of feature detecting `console`.

Reply:
done via deprecatedWarn